### PR TITLE
Benchmark setup

### PR DESCRIPTION
**Pull request by debashishc**
### Pull Request: Initial Skeleton Setup for CUDA Matrix Multiplication Benchmark

This PR sets up the basic skeleton for the CUDA matrix multiplication benchmark project, including directory structure, initial files, and placeholder content for kernel implementations and benchmarking scripts.

- Organized the project into a modular structure with directories for `src/`, `benchmarks/`, and `include/`.
- Added initial CUDA kernel files:
  - `mm_basic.cu`: Basic matrix multiplication (in progress).
  - `mm_shared_memory.cu`: Shared memory optimized kernel (in progress).
  - `mm_tiled.cu`: Tiled matrix multiplication (in progress).
- Created `main.cu` as the entry point for running benchmarks (skeleton).
- Included a `Makefile` for automating the build process (basic version).
- Added a benchmark script (`run_benchmarks.sh`) to run future benchmarks (skeleton).
- Updated the `README.md` to reflect the current project structure and status (marked as DRAFT).

**Next Steps**:
- Complete the implementation of the matrix multiplication kernels.
- Add matrix utilities and benchmarking functionality.
- Refine the `run_benchmarks.sh` script to support kernel selection and size parameters.

### CODE REVIEW

File: how_to/benchmark_cuda_matmul/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+Ensure you have the following installed:
+
+- **Hardware**: CUDA-enabled GPU
+- **Software**:
+  - NVIDIA CUDA Toolkit
+  - GNU Make
+  - Bash shell (for running scripts)
+  - Python 3 (for plotting results in future updates)
+  - Python libraries: `matplotlib`, `numpy` (for future visualizations)
```
Comment (sarcosholzinger) [inline how_to/benchmark_cuda_matmul/README.md:60]: I'd suggest using Plotly over matplotlib

File: how_to/benchmark_cuda_matmul/README.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+cd benchmarks/
+chmod +x run_benchmarks.sh
+./run_benchmarks.sh
+```
+
+This script will execute different matrix multiplication kernels across various matrix sizes and store the results in the `results/` directory.
+
+---
+
+## Implementations Included
```
Comment (sarcosholzinger) [inline how_to/benchmark_cuda_matmul/README.md:87]: Adding version (SemVer) tagging would be useful.

File: how_to/benchmark_cuda_matmul/src/include/matrix_utils.h
Comment (sarcosholzinger) [inline how_to/benchmark_cuda_matmul/src/include/matrix_utils.h:1]: note- empty file.

File: how_to/benchmark_cuda_matmul/src/kernels/mm_basic.cu
Comment (sarcosholzinger) [inline how_to/benchmark_cuda_matmul/src/kernels/mm_basic.cu:1]: note- empty file.

File: how_to/benchmark_cuda_matmul/src/kernels/mm_shared_mem.cu
Comment (sarcosholzinger) [inline how_to/benchmark_cuda_matmul/src/kernels/mm_shared_mem.cu:1]: note- empty file.

File: how_to/benchmark_cuda_matmul/src/kernels/mm_tiled.cu
Comment (sarcosholzinger) [inline how_to/benchmark_cuda_matmul/src/kernels/mm_tiled.cu:1]: note- empty file.

File: how_to/benchmark_cuda_matmul/src/main.cu
Comment (sarcosholzinger) [inline how_to/benchmark_cuda_matmul/src/main.cu:1]: note- empty file.

**Review by sarcosholzinger — COMMENTED** [review]
Added a few suggestions and noted a few empty files for your acknowledgement.

File: how_to/benchmark_cuda_matmul/src/main.cu
Comment (debashishc) [reply how_to/benchmark_cuda_matmul/src/main.cu:1]: file left empty for git commit purposes. empty directory is not included in a git commit.

File: how_to/benchmark_cuda_matmul/README.md (hunk 1 above)
Comment (debashishc) [reply how_to/benchmark_cuda_matmul/README.md:60]: thanks, I ll consider Plotly in my plots

**Review by sarcosholzinger — APPROVED** [review]

**Pull request merged**